index.bs

```diff
@@ -3701,6 +3727,792 @@ opened</dfn> steps given |window|, |type| and |message|.
+   <code>headers</code> field set to |headers|, the |cookies| field set to
+   |cookies|, and the <code>headersSize</code> field set to |headers size|,
+   the <code>bodySize</code> field set to |body size|, and the
+   <code>timings</code> field set to |timings|.
+
+</div>
+
+#### The network.ResponseContent type #### {#type-network-ResponseContent}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.ResponseContent = {
+    size: unit / null
+};
+</pre>
+
+The <code>network.ResponseContent</code> type represents the decoded response to
+a network request.
+
+<!-- Not sure this is worthwhile if it only ends up with a single field, but it
+would be natural to add a field here if we have a way to return the body -->
+
+<div algorithm>
+To <dfn>get the response content info</dfn> given |response|.
+
+1. Return a new map matching the <code>network.ResponseContent</code>
+   production, with the <code>size</code> field set to |response|'s [=response body
+   info=]'s [=decoded size=]
+
+</div>
+
+#### The network.ResponseData type #### {#type-network-ResponseData}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.ResponseData = {
+    url: text,
+    protocol: text,
+    status: uint,
+    statusText: text,
+    fromCache: bool,
+    headers: [*network.Header],
+    mimeType: text,
+    bytesReceived: uint,
+    headersSize: uint / null,
+    bodySize: uint / null,
+    content: network.ResponseContent
+};
+</pre>
+
+The <code>network.RequestData</code> type represents the response to a network
+request.
+
+<div algorithm>
+
+To <dfn>get the protocol</dfn> given |response|:
+
+1. Let |protocol| be the empty string.
+
+1. If |response|'s [=final connection timing info=] is not null, set |protocol|
+   to |response|'s [=final connection timing info=]'s [=ALPN negotiated
+   protocol=].
+
+1. If |protocol| is the empty string, or is equal to "<code>unknown</code>":
+
+  1. Set |protocol| to |response|'s [=response/url=]'s [=url/scheme=]
+
+  1. If |protocol| is equal to either "<code>http</code>" or
+     "<code>https</code>" and |response| has an associated HTTP Response.
+
+     Note: [[FETCH]] isn't clear about the relation between a HTTP network response and
+     a response object.
+
+     <!-- It would be better to move this into Fetch itself. -->
+
+    1. Let |http version| be the HTTP Response's Status line's HTTP-version [[HTTP11]].
+
+    1. If |http version| starts with "<code>HTTP/</code>":
+
+      1. Let |version| be the [=code unit substring=] of |HTTP version| from 5
+         to |http version|'s [=string/length=].
+
+     1. If |version| is "<code>0.9</code>", set |protocol| to
+        "<code>http/0.9</code>", otherwise if |version| is "<code>1.0</code>",
+        set |protocol| to "<code>http/1.0</code>", otherwise if |version| is
+        "<code>1.1</code>", set |protocol| to "<code>http/1.1</code>".
```

Comment:
```suggestion
      1. If |version| is "<code>0.9</code>", set |protocol| to
         "<code>http/0.9</code>", otherwise if |version| is "<code>1.0</code>",
         set |protocol| to "<code>http/1.0</code>", otherwise if |version| is
         "<code>1.1</code>", set |protocol| to "<code>http/1.1</code>".
```

wrong indentation